src/index.ts

```diff
@@ -71,17 +71,37 @@ export function resolveBundle(bundle: Bundle, resolvers: Resolver[]): Promise<Re
 			// Now that we have a resolver, add the new files needed to the stream
 			resolver.resolve(bundle)
 			.then((additionalItems) => {
-				additionalItems.map((file: FileInfo) => {
+
+				return Promise.map(additionalItems, (file: FileInfo) => {
 					pack.entry({ name: file.name, size: file.size }, file.contents)
+					if (file.name === 'Dockerfile') {
+						return bundle.dockerfileHook(file.contents.toString())
+					} else {
+						return Promise.resolve()
+					}
+				})
+				.then(() => {
+
+					return Promise.try(() => {
+						if (resolver.name === 'Standard Dockerfile') {
+							// The hook will not have been ran on this file yet, as the Dockerfile was not added
+							// as an additional item
+							return bundle.dockerfileHook((resolver as DockerfileResolver).getDockerfileContents())
+						}
+						return;
+					});
+
 				})
+				.then(() => {
 
-				// all of the extra files have now been added to the stream, resolve
-				// the promise with it
-				pack.finalize()
+					// all of the extra files have now been added to the stream, resolve
+					// the promise with it
+					pack.finalize()
```

Comment:
Didn't the linter notify of the wrong indentation before? We definitely need that